PocketKit/Sources/PocketKit/Notifications/InstantSync.swift

```diff
@@ -0,0 +1,138 @@
+        self.source.refresh {
+            completionHandler(.newData)
+        }
+        self.source.retryImmediately()
+    }
+}
+
+/**
+ Conforming InstantSync to the InstantSyncProtocol
+ */
+extension InstantSync: InstantSyncProtocol {
+
+    func loggedIn(session: SharedPocketKit.Session) {
+        // Instant sync does not need to do anything on login, we only register tokens when we recieve one.
+        // This is to conform to the APNS protocol
+    }
+
+    func loggedOut(session: SharedPocketKit.Session?) {
+        // Check if we got a session from our caller
+        guard let session = session else {
+            // We do not have a session so we can not deregister for push notifications.
+            // This is likely because the app opened to the login screen where we would have not had a session beforehand.
+            return
+        }
+
+        Task {[weak self] in
+            do {
+                _ = try await self?.v3Client.deregisterPushToken(for: DeviceUtilities.deviceIdentifer(), pushType: pushType, session: session)
+            } catch {
+                Crashlogger.capture(error: error)
+            }
+        }
+    }
+
+    func register(deviceToken: Data) {
+        // Check if we have a current session
+        guard let session = appSession.currentSession else {
+            // We do not have a session so we can not register for push notifications.
+            // Capture the message for later use and move on.
+            Crashlogger.capture(message: "Push Notification Service has no current session to use to register a push token with")
+            return
+        }
+
+        Task {[weak self] in
+            do {
+                _ = try await self?.v3Client.registerPushToken(for: DeviceUtilities.deviceIdentifer(), pushType: pushType, token: deviceToken.base64EncodedString(), session: session)
+            } catch {
+                Crashlogger.capture(error: error)
+            }
+        }
+    }
+
+    /**
+     Recieved a background notification
+     */
+    func handleBackgroundNotifcation(didReceiveRemoteNotification userInfo: [AnyHashable: Any], fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void) -> Bool {
+        self.handleInstantSync(didReceiveRemoteNotification: userInfo, fetchCompletionHandler: completionHandler)
+        return true
+    }
+
+    /**
+     Handle our Instant Sync push notification
+     */
+    func handleInstantSync(
+        didReceiveRemoteNotification userInfo: [AnyHashable: Any],
+        fetchCompletionHandler completionHandler: @escaping (UIBackgroundFetchResult) -> Void
+    ) {
+        guard let session = appSession.currentSession else {
+            // The user is not logged in, so we ignore the instant sync.
+            completionHandler(.noData)
+            return
+        }
+
+        guard let pushGuid = userInfo["g"] as? String else {
```

Comment:
🤣  Legacy ios code :)  It will not change for the foreseeable future. G is actually a user's guid, it is rarely populated but it may be from legacy systems.